src/vs/workbench/contrib/webviewPanel/browser/webviewPanel.contribution.ts

```diff
@@ -27,21 +26,32 @@ import { IWebviewWorkbenchService, WebviewEditorService } from './webviewWorkben
 	localize('webview.editor.label', "webview editor")),
 	[new SyncDescriptor(WebviewInput)]);
 
-class WebviewPanelContribution implements IWorkbenchContribution {
+class WebviewPanelContribution extends Disposable implements IWorkbenchContribution {
+
+	private readonly _disposables = new DisposableStore();
 	constructor(
-		@IEditorService private readonly editorService: IEditorService,
 		@IEditorGroupsService private readonly editorGroupService: IEditorGroupsService,
 	) {
-		this.editorService.overrideOpenEditor({
-			open: (editor, options, group) => this.onEditorOpening(editor, options, group)
-		});
+		super();
+		// Add all the initial groups to be listened to
+		this.editorGroupService.whenReady.then(() => this.editorGroupService.groups.forEach(group => {
+			this.registerGroupListener(group);
+		}));
+
+		// Additional groups added should also be listened to
+		this._register(this.editorGroupService.onDidAddGroup((group) => this.registerGroupListener(group)));
+
+		this._register(this._disposables);
 	}
 
+	private registerGroupListener(group: IEditorGroup): void {
+		const listener = group.onWillOpenEditor(e => this.onEditorOpening(e.editor, group));
```

Comment:
This does not free up group listeners when groups close, which is not ideal. But I am thinking of maybe providing a better listener from the service that makes sure all groups are listened to.